### ISSUE TITLE
Does not report Incompatible Method Override when override overload

### ISSUE THREAD

**Issue by juk-r** [microsoft/pyright#6382]
In this example:
```python
from typing import overload

class A: ...
class B: ...
class C: ...

class Sup:
    @overload
    def fun(self, x: A) -> A: ...
    @overload
    def fun(self, x: B) -> B: ...
    def fun(self, x: A | B) -> A | B: ...
    
class Sub(Sup):
    @overload
    def fun(self, x: C) -> C: ...
    @overload
    def fun(self, x: B) -> B: ...
    def fun(self, x: C | B) -> C | B:
        # must report here
        ...

```
using pyright 1.1.335

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the issue and confirm it's a bug. I'll investigate further.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
